microsoft-edge/progressive-web-apps-chromium/how-to/offline.md

```diff
@@ -1,183 +1,155 @@
+IndexedDB is an asynchronous API for storing structured data that can be used in your app's front-end code or service worker code. Use the IndexedDB API for storing a significant amount of structured data on the client, or binary data, such as encrypted media objects or files.
 
+IndexedDB is the best option for storing data in your PWA because using the API does not slow down your app by blocking the main thread, and it can be used both from your app's front-end code and service worker.
 
-<!-- ====================================================================== -->
-## Understand storage options for PWAs
+Using IndexedDB is more complex than using Web Storage and requires the following steps to store data:
 
-Sometimes you might need to store small amounts of data in order to provide a better offline experience for your users. If that is the case, you might find that the simplicity of the key-value pair system of Web Storage meets your needs.
+1. Opening a database by using the `window.indexedDB.open()` function.
+1. Creating an object store in the database by using the `IDBDatabase.createObjectStore()` function.
+1. Starting a transaction to store data by using the `IDBDatabase.transaction()` function.
+1. And waiting for the operation to complete by listening to an event.
 
-> [!IMPORTANT]
-> Web Storage is a synchronous process, and isn't available for use within worker threads, such as Service Workers. Heavy usage of Web Storage may create performance issues for your application.
+To learn more and view code examples, see [Using IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API/Using_IndexedDB) on MDN.
 
-There are two types of Web Storage: `localStorage` and `sessionStorage`. Each type of Web Storage is maintained as a separate data store that's isolated to the domain that created it.
 
-*  `sessionStorage` persists only for the duration of the browsing session. For example, while the browser is open, which includes refresh and restores.
-*  `localStorage` persists until the data is removed by the code, the user, or the browser. For example, when there is limited storage available.
+<!-- ====================================================================== -->
+## Cache
 
-The following code shows how to use `localStorage`, which is similar to how `sessionStorage` is used:
+The Cache API is a system for storing and retrieving network requests and responses in your app's front-end code or service worker. It can be used to store assets, such as images and files, locally on the user's device, which can make your application work offline or improve its performance by reducing the number of network requests needed to render the app.
```

Comment:
```suggestion
The Cache API is a system for storing and retrieving network requests and responses in your app's front-end code or service worker. It can be used to store assets, such as images and files, locally on the user's device. This can make your application work even when it's offline, or improve its performance by reducing the number of network requests that are needed to render the app.
```